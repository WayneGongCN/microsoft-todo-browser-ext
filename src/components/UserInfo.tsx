import React from 'react';
import { useSelector } from 'react-redux';
import Avatar from '@material-ui/core/Avatar';
import { State } from '../redux';
import { Container, Grid } from '@material-ui/core';

const UserInfo: React.FC = () => {
  const account = useSelector((state: State) => state.auth.authenticationResult?.account);

  if (!account) return null;
  return (
    <Grid container alignItems='center' spacing={1}>
      <Grid item>
        <Avatar>{account.username.slice(0, 2)}</Avatar>
      </Grid>
      <Grid item>{account.username}</Grid>
    </Grid>
  );
};

export default UserInfo;
