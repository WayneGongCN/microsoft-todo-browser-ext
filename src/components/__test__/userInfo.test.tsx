/* eslint-disable @typescript-eslint/ban-ts-comment */
import React from 'react';
import snapshotRender from 'react-test-renderer';
import UserInfo from '../UserInfo';
import { useSelector } from 'react-redux';


jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));
describe('Snapshot test', () => {
  test('normal render', () => {
    // @ts-ignore
    useSelector.mockReturnValue({
      username: "wayne.gong.cn@hotmail.com",
    })
    const tree = snapshotRender.create(<UserInfo />).toJSON();
    expect(tree).toMatchSnapshot('normal');
  });
});
