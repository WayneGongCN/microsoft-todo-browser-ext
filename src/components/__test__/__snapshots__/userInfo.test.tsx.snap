// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test normal render: normal 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-7ldjcq-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
    >
      wa
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    wayne.gong.cn@hotmail.com
  </div>
</div>
`;
